gplugins/lumerical/utils.py

```diff
@@ -0,0 +1,154 @@
+    """
+    layerstack = layerstack or get_layer_stack()
+
+    layer_builder = Element("layer_builder")
+
+    process_name = SubElement(layer_builder, "process_name")
+    process_name.text = "process"
+
+    layers = SubElement(layer_builder, "layers")
+    doping_layers = SubElement(layer_builder, "doping_layers")
+    for layer_name, layer_info in layerstack.to_dict().items():
+        ### Set optical and metal layers
+        if layer_info["layer_type"] == "grow":
+            process = "Grow"
+        elif layer_info["layer_type"] == "background":
+            process = "Background"
+        elif layer_info["layer_type"] == "doping":
+            process = "Implant"
+        else:
+            logger.warning(
+                f'"{layer_info["layer_type"]}" layer type not supported for "{layer_name}" in Lumerical. Skipping in LBR process file generation.'
+            )
+            process = "Grow"
+
+        if process == "Grow" or process == "Background":
+            layer = SubElement(layers, "layer")
+
+            # Default params
+            layer.set("enabled", "1")
+            layer.set("pattern_alpha", "0.8")
+            layer.set("start_position_auto", "0")
+            layer.set("background_alpha", "0.3")
+            layer.set("pattern_material_index", "0")
+            layer.set("material_index", "0")
+
+            # Layer specific params
+            layer.set("name", layer_name)
+            layer.set(
+                "layer_name", f'{layer_info["layer"][0]}:{layer_info["layer"][1]}'
+            )
+            layer.set("start_position", f'{layer_info["zmin"] * um}')
+            layer.set("thickness", f'{layer_info["thickness"] * um}')
+            layer.set("start_position", f'{layer_info["zmin"] * um}')
+            layer.set("process", f"{process}")
+            layer.set("sidewall_angle", f'{90-layer_info["sidewall_angle"]}')
+            if layer_info["bias"]:
+                layer.set("pattern_growth_delta", f"{layer_info['bias'] * um}")
+            else:
+                layer.set("pattern_growth_delta", "0")
+
+            if process == "Grow":
+                layer.set(
+                    "pattern_material",
+                    f'{material_map.get(layer_info["material"], "")}',
+                )
+            elif process == "Background":
+                layer.set("material", f'{material_map.get(layer_info["material"], "")}')
+
+        if process == "Implant" or process == "Background":
+            ### Set doping layers
+            # KNOWN ISSUE: If a metal or optical layer has the same name as a doping layer, Layer Builder will not compile
+            # the process file correctly and the doping layer will not appear. Therefore, doping layer names MUST be unique.
+            # FIX: Appending "_doping" to name
+
+            # KNOWN ISSUE: If the 'process' is not 'Background' or 'Implant', this will crash CHARGE upon importing process file.
```

Comment:
is this just for dopings?

Reply:
Yes, doping layers seem to have an issue with the other layers. Will need to report this Lumerical in the future, but this will be our current workaround.

Reply:
I think I understand now- you can only set doping parameters for background/implant layers? that makes sense to me- where's the issue?

Reply:
Line 98: Doping layers in the lbr file can only have process specified as "Background" or "Implant". "Grow" does not make sense for doping layers, so "Grow" is not supported.

Reply:
"grow" shouldn't be a doping layer, right? is your point that lumerical should error (rather than crash) here?

Reply:
Yes. In gdsfactory.generic_tech.LAYER_STACK, the substrate has background_doping_concentration with process set to "grow". When exporting to lbr then importing into Lumerical, Lumerical crashes.